internal/charmstore/migrations.go

```diff
@@ -175,12 +175,12 @@ func addDevelopmentACLs(db StoreDatabase) error {
 	baseEntities := db.BaseEntities()
 	var baseEntity mongodoc.BaseEntity
 	iter := baseEntities.Find(bson.D{{
-		"developmentacls", bson.D{{"$exists", false}},
-	}}).Select(bson.D{{"_id", 1}, {"acls", 1}}).Iter()
+		"channelacls.development", bson.D{{"$exists", false}},
+	}}).Select(bson.D{{"_id", 1}, {"channelacls", 1}}).Iter()
 	defer iter.Close()
 	for iter.Next(&baseEntity) {
 		if err := baseEntities.UpdateId(baseEntity.URL, bson.D{{
-			"$set", bson.D{{"developmentacls", baseEntity.ACLs}},
+			"$set", bson.D{{"channelacls.development", baseEntity.ChannelACLs[mongodoc.DevelopmentChannel]}},
```

Comment:
I'm not sure what we should do about migrations in the face of field renames.
Technically each migration step should not need to change because of
a later step. Perhaps we'd be better off keeping all the old mongodoc types
around.

Alternatively, if we're sure that all the charmstore instances that we care about
have migrated their database with this step already, we could just delete
it.